app/controllers/games_controller.rb

```diff
@@ -22,16 +22,38 @@ def create
   end
 
   def update
-    @game = Game.find(params[:id])
-    if current_user.id == params[:jedi_user_id].to_i
-      @game.update(sith_user_id: current_user.id)
-      redirect_to ships_path
-    elsif current_user.id == params[:sith_user_id].to_i
-      @game.update(jedi_user_id: current_user.id)
-      redirect_to ships_path
-    else
-      render text: "Invalid Request", status: :unprocessable_entity
-    end
+    valid_update
+    # begin
+    #   valid_update
+    # rescue ActiveRecord::RecordInvalid
+    #   redirect_to games_path, flash: {alert: current_game.errors.full_messages.last} 
+    # rescue
+    #   redirect_to games_path, flash: {alert: "Invalid Request."} 
+    # end
   end
+
+  private
+
+  helper_method :current_game
   
+  def current_game
+    @current_game ||= Game.find(params[:id])
+  end
+
+  def game_params
+    params.require(:game).permit(:jedi_user_id, :sith_user_id)
+  end
+
+  def valid_update
+    # current_game.errors.clear
+    if current_game.sith_user_id.nil?
+      current_game.update_attributes!(sith_user_id: current_user.id)
+      redirect_to ships_path, flash: {
+        notice: "You joined the game! Please place your ships."}
+    else current_game.jedi_user_id.nil?
```

Comment:
I think this should be `elsif` which I think is why the test wasn't working below.